Framework/API/src/ComponentInfo.cpp

```diff
@@ -12,17 +13,19 @@ namespace API {
 
 /**
  * Constructor
- * @param componentInfo : Beamline wrapped ComponentInfo
- * @param componentIds : Component Ids ordered by component index
+ * @brief ComponentInfo::ComponentInfo
+ * @param componentInfo
+ * @param componentIds : ComponentIDs ordered by component
+ * index
  */
-ComponentInfo::ComponentInfo(const Beamline::ComponentInfo &componentInfo,
-                             std::vector<Geometry::ComponentID> &&componentIds)
+ComponentInfo::ComponentInfo(
+    Mantid::Beamline::ComponentInfo &componentInfo,
+    std::vector<Mantid::Geometry::IComponent *> componentIds)
```

Comment:
Do you mean ref to `Beamline::ComponentInfo`? Now needs to be non-const so that `setPosition`, `setRotation` can be propagated